src/components/vm/snapshots/vmSnapshotsCreateModal.jsx

```diff
@@ -175,19 +239,25 @@ export class CreateSnapshotModal extends React.Component {
         }
     }
 
+    componentDidMount() {
+        this.updateAvailableSpace(this.state.memoryLocation);
+    }
+
     render() {
         const Dialogs = this.context;
         const { idPrefix, isExternal, vm } = this.props;
-        const { name, description, memoryPath } = this.state;
+        const { name, description, memoryLocation, available } = this.state;
         const validationError = this.onValidate();
 
         const body = (
             <Form onSubmit={e => e.preventDefault()} isHorizontal>
                 <NameRow name={name} validationError={validationError.name} onValueChanged={this.onValueChanged} />
                 <DescriptionRow description={description} onValueChanged={this.onValueChanged} />
                 {isExternal && vm.state === 'running' &&
-                    <MemoryPathRow memoryPath={memoryPath} onValueChanged={this.onValueChanged}
-                                   validationError={validationError.memory} />}
+                    <MemoryLocationRow memoryLocation={memoryLocation} onValueChanged={this.onValueChanged}
+                                       validationError={validationError.memory}
+                                       available={available}
+                                       needed={(vm.rssMemory || vm.currentMemory) * 1024} />}
```

Comment:
Did you measure this for a few instances, is that accurate? Or does it need some factor, either > 1 for some overhead, or even < 1 because it gets compressed or created as sparse file or so?

Reply:
> Did you measure this for a few instances, is that accurate?

I did measure it, but I was a while and I should do it again and record the results.